Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -86,13 +86,38 @@ class TIFFLoadingContext {
         return value << (8 - bits);
     }
 
+    u8 number_of_unknown_channel() const
+    {
+        switch (*m_metadata.photometric_interpretation()) {
+        case PhotometricInterpretation::WhiteIsZero:
+        case PhotometricInterpretation::BlackIsZero:
+            return *m_metadata.samples_per_pixel() - 1;
+        case PhotometricInterpretation::RGB:
+            return *m_metadata.samples_per_pixel() - 3;
+        default:
+            TODO();
+        }
+    }
+
     ErrorOr<Color> read_color(BigEndianInputBitStream& stream)
     {
         auto bits_per_sample = *m_metadata.bits_per_sample();
+
+        // Section 7: Additional Baseline TIFF Requirements
+        // Some TIFF files may have more components per pixel than you think. A Baseline TIFF reader must skip over
+        // them gracefully, using the values of the SamplesPerPixel and BitsPerSample fields.
+        auto discard_unknown_channels = [&]() -> ErrorOr<void> {
+            auto const unknown_channels = number_of_unknown_channel();
```

Comment:
nit: How about making the function return `samples_for_photometric_interpretation()` and say `unknown_channels = samples_per_pixel() - samples_for_photometric_interpretation(*m_metadata.photometric_interpretation()` here?